### PR TITLE
Clarify logger flag

### LINKED ISSUE #7189: Unclear default Logger
## 📚 Documentation

The logger flag for `Trainer` does not have documentation for behavior of the boolean value inputs. Adding description of this performance to the docs.

### PR DESCRIPTION
**Pull request by SpontaneousDuck**
Clarify behavior of boolean values on the logger flag for Trainer.

## What does this PR do?
Adds documentation for boolean values input to the logger flag of Trainer.

Fixes #7189 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Easy doc edit! 🐱‍👤

### CODE REVIEW

File: docs/source/common/trainer.rst [hunk 1]
```diff
@@ -870,7 +870,7 @@ logger
 
 |
 
-:doc:`Logger <../common/loggers>` (or iterable collection of loggers) for experiment tracking.
+:doc:`Logger <../common/loggers>` (or iterable collection of loggers) for experiment tracking. A `True` value uses the default `TensorBoardLogger` shown below. `False` will disable logging.
```
Comment (ethanwharris) [inline docs/source/common/trainer.rst:873]:
```suggestion
:doc:`Logger <../common/loggers>` (or iterable collection of loggers) for experiment tracking. A ``True`` value uses the default ``TensorBoardLogger`` shown below. ``False`` will disable logging.
```

**Review by ethanwharris — APPROVED** [review]
Thanks for the PR, looks good to me :smiley:

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
Could you also add this to the trainer docstrings in `trainer.py`?

**Review by ananthsub — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

File: docs/source/common/trainer.rst [hunk 2]
```diff
@@ -870,7 +870,7 @@ logger
 
 |
 
-:doc:`Logger <../common/loggers>` (or iterable collection of loggers) for experiment tracking.
+:doc:`Logger <../common/loggers>` (or iterable collection of loggers) for experiment tracking. A ``True`` value uses the default ``TensorBoardLogger`` shown below. ``False`` will disable logging.
```
Comment (tchaton) [inline docs/source/common/trainer.rst:873]:
```suggestion
:doc:`Logger <../common/loggers>` (or iterable collection of loggers) for experiment tracking. 

.. note:: 

       Using ``Trainer(logger=True)`` will use the ``TensorBoardLogger`` by default and ``Trainer(logger=False)`` will disable logging.
```
